### PR TITLE
rework pipeline

### LINKED ISSUE #110: Rework of the pipelines - Serverless
## Feature Request

**Description:**

We propose to enhance our CI/CD pipeline by adopting the updated workflow depicted in the pipeline deployment diagram : https://www.figma.com/file/TWa2BHTdyIVo0TzRSJOVd9/Pipeline---API?type=whiteboard&node-id=0%3A1&t=kFx8743zic1pgGs7-1

Introduces a robust 'Dev Pipeline' with enhanced steps for automated linting, unit testing, integration testing, static analysis, and security scanning before the AWS build & deploy stage. 

**Use Case:**

The updated pipeline is beneficial to both the development team and the stakeholders for the following reasons:

- Developers will have immediate feedback on potential issues, enabling them to fix problems earlier in the development cycle.
- Reviewers will have automated checks to assist in the code review process, ensuring that only high-quality code is merged into the main branch.

**Proposed Solution:**

The pipeline update should include:

- Unit Testing: To ensure that individual components work as expected.
- Static Analysis: To identify potential security vulnerabilities and maintain code quality.

**Alternative Solutions:**

**Additional Information:**

Automated testing is not define yet.

## Checklist

- [x] I have searched for existing feature requests that are similar.
- [x] I have provided a clear and concise description of the feature.
- [x] I have explained the use case for the feature.
- [x] I have considered possible implementation approaches.

### PR DESCRIPTION
**Pull request by CharlesBourgeois**
## Description

We propose to enhance our CI/CD pipeline by adopting the updated workflow depicted in the pipeline deployment diagram : https://www.figma.com/file/TWa2BHTdyIVo0TzRSJOVd9/Pipeline---API?type=whiteboard&node-id=0%3A1&t=kFx8743zic1pgGs7-1

## Changes Made

Introduces a robust 'Dev Pipeline' with enhanced steps for automated linting, unit testing, integration testing, static analysis, and security scanning before the AWS build & deploy stage.

## Related Issue

#110 

## Checklist

- [x] I have tested these changes locally.
- [x] I have included necessary documentation updates (if applicable).
- [x] My code follows the project's coding standards.
- [x] All existing tests are passing.
- [x] I have added new tests (if new features or changes warrant them).

### CODE REVIEW

File: .github/actions/run-tests/action.yml [hunk 1]
```diff
@@ -1,17 +1,10 @@
...
-    - name: Run tests
-      env:
-        API_URL: mockURL
-        API_KEY: mockKey
-      run: npm run test 
+  steps:
+    - run: npm install -g n
+      shell: bash
+    - run: npm run test
       shell: bash
```
Comment (k-allagbe) [inline .github/actions/run-tests/action.yml]: missing end of file

File: .github/workflows/on-pr-workflow.yml [hunk 2]
```diff
@@ -2,19 +2,30 @@ name: Continuous Integration
...
+  lint-and-test:
+    name: Lint, test and deploy test coverage to Github Pages
+    uses: ./.github/workflows/lint-test-workflow.yml
+    with:
+      deploy: ${{ github.event.inputs.enable_deployment }}
 
-      
+  security-and-analysis:
+    name: Security and code analysis
+    uses: ./.github/workflows/security-analysis-workflow.yml
```
Comment (k-allagbe) [inline .github/workflows/on-pr-workflow.yml]: missing end of file. Could maybe be automated.

**Review by k-allagbe — COMMENTED** [review]
Just missing some end of files. Otherwise, 🚀

File: .github/workflows/on-pr-workflow.yml (hunk 2 above)
Comment (CharlesBourgeois) [reply .github/workflows/on-pr-workflow.yml]: already automated, but only for source files (too much permissions problem with other files)

**Review by k-allagbe — APPROVED** [review]

**Pull request merged**